### PR TITLE
[16.0][MIG] stock_request_mrp: Migration to 16.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 16.0

Please @pedrobaeza and @carolinafernandez-tecnativa can you review it?

@Tecnativa TT47866

### CODE REVIEW

**Review by carolinafernandez-tecnativa — APPROVED** [review]
Functional review LGTM, Thanks :)

File: stock_request_mrp/tests/test_stock_request_mrp.py [hunk 1]
```diff
@@ -92,13 +92,11 @@ def test_create_request_01(self):
         self.assertEqual(
             manufacturing_order.company_id, order.stock_request_ids[0].company_id
         )
-        manufacturing_order.action_confirm()
-        manufacturing_order.write(
-            {"qty_producing": manufacturing_order.product_uom_qty}
+        res = manufacturing_order.button_mark_done()
```
Comment (pedrobaeza) [inline stock_request_mrp/tests/test_stock_request_mrp.py:96]: Why putting the production as done? That's different from confirming it, which was the previous action.

File: stock_request_mrp/views/mrp_production_views.xml [hunk 2]
```diff
@@ -18,18 +14,27 @@
                     class="oe_stat_button"
                     icon="fa-chain"
                     attrs="{'invisible':[('stock_request_count', '=', 0)]}"
+                    groups="stock_request.group_stock_request_user"
                 >
                     <field
                         name="stock_request_count"
                         widget="statinfo"
                         string="Stock Requests"
+                        groups="stock_request.group_stock_request_user"
```
Comment (pedrobaeza) [inline stock_request_mrp/views/mrp_production_views.xml]: No need of group in this nested one

File: stock_request_mrp/views/stock_request_order_views.xml [hunk 3]
```diff
@@ -6,18 +6,27 @@
...
                     icon="fa-wrench"
                     attrs="{'invisible': [('production_count', '=', 0)]}"
+                    groups="mrp.group_mrp_user"
                 >
-                    <field name="production_count" widget="statinfo" string="MOs" />
+                    <field
+                        name="production_count"
+                        widget="statinfo"
+                        string="MOs"
+                        groups="mrp.group_mrp_user"
```
Comment (pedrobaeza) [inline stock_request_mrp/views/stock_request_order_views.xml]: No need

File: stock_request_mrp/views/stock_request_views.xml [hunk 4]
```diff
@@ -6,18 +6,27 @@
...
                     icon="fa-wrench"
                     attrs="{'invisible': [('production_count', '=', 0)]}"
+                    groups="mrp.group_mrp_user"
                 >
-                    <field name="production_count" widget="statinfo" string="MOs" />
+                    <field
+                        name="production_count"
+                        widget="statinfo"
+                        string="MOs"
+                        groups="mrp.group_mrp_user"
```
Comment (pedrobaeza) [inline stock_request_mrp/views/stock_request_views.xml]: No need

File: stock_request_mrp/tests/test_stock_request_mrp.py (hunk 1 above)
Comment (victoralmau) [reply stock_request_mrp/tests/test_stock_request_mrp.py:96]: Mark as done to finalize and check `qty_done` (same as in v15: https://github.com/OCA/stock-logistics-warehouse/blob/15.0/stock_request_mrp/tests/test_stock_request_mrp.py#L101).

Comment (pedrobaeza) [reply stock_request_mrp/tests/test_stock_request_mrp.py:96]: OK, and no need to previously confirm it?

Comment (victoralmau) [reply stock_request_mrp/tests/test_stock_request_mrp.py:96]: It is not necessary to confirm it (in v15 it is not really necessary either); however, I have added in the test to check that the production is confirmed.

**Comment by victoralmau** [comment]
Changes done.

File: stock_request_mrp/views/stock_request_views.xml (hunk 4 above)
Comment (pedrobaeza) [reply stock_request_mrp/views/stock_request_views.xml]: This one hasn't been changed yet

**Review by pedrobaeza — COMMENTED** [review]
Please squash the latest commit into the migration one.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**